server/odc-service/src/main/java/com/oceanbase/odc/service/connection/database/DatabaseService.java

```diff
@@ -620,7 +599,7 @@ private Database entityToModel(DatabaseEntity entity) {
             model.setProject(projectService.detail(entity.getProjectId()));
         }
         model.setDataSource(connectionService.getForConnectionSkipPermissionCheck(entity.getConnectionId()));
-        model.setEnvironment(environmentService.detailSkipPermissionCheck(entity.getEnvironmentId()));
+        model.setEnvironment(environmentService.detailSkipPermissionCheck(model.getDataSource().getEnvironmentId()));
```

Comment:
It looks like there's no way we can save this IO cost. entityToModel is only used in the DatabaseService#detail and DatabaseService#create, which I think is acceptable and actually, in the real scene, the create database and the update database APIs' RT look like normal.
On the other hand, in the situation of batch operation, we use another method entitiesToModels and this method loads other POJOs in one time rather than in iteration one by one, so there is no additional time cost as well.